sabnzbd/filesystem.py

```diff
@@ -166,7 +166,19 @@ def sanitize_filename(name: str) -> str:
     if not name:
         name = "unknown"
 
+    # now split name into name, ext
     name, ext = os.path.splitext(name)
+
+    # If filename is too long, truncate it:
+    if len(name) + len(ext) > 222:
+        logging.debug("Filename %s is too long, so truncating", name + ext)
+        # Too long filenames are often caused by incorrect non-ascii chars,
+        # so brute-force remove those non-ascii chars, and only keep 222 chars
+        # ... keeping in place the original extension
+        name = str(name.encode("ascii", "ignore"), "utf-8")
+        name = ''.join( c for c in name if  ord(c) >= 32 and ord(c)<128 )
+        name = name[: 222 - len(ext)]
```

Comment:
The length of the extension isn't sanity-checked anywhere. With os.path.splitext simply returning everything after the last dot, you're in for unexpected results for the value `name` once `222 - len(ext)` goes negative - both longer and (much) shorter than intended.

It would make sense to add tests guarding against such issues as well.